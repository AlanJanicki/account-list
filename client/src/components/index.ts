export { Card } from './Card/Card';
export { Loader } from './Loader/Loader';
export { Spinner } from './Spinner/Spinner';
export * from './Table/Table';
