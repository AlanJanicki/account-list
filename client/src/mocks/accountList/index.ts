import { v4 as uuidv4 } from 'uuid';

import { HEADERS } from '../../features';
import { transformToArrayOfObjectsWithGivenKeys } from '../../utils';
import accounts from './accounts.json';

export const getCellsWithMockedData = () => {
  const headers = Object.values(HEADERS).map(({ key }) => key);

  return [
    ...headers.map((value) => ({ isHeader: true, key: uuidv4(), value })),
    ...transformToArrayOfObjectsWithGivenKeys({
      data: accounts.slice(0, 3),
      keys: headers,
      valueFallback: '-'
    })
  ];
};
