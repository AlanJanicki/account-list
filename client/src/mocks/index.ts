export * from './accountList';
export { default as accounts } from './accountList/accounts.json';
export { default as accountTypes } from './accountList/accounttypes.json';
export * from './msw';
