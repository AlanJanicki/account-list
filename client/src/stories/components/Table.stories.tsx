import { Table } from '../../components';
import { getCellsWithMockedData } from '../../mocks/accountList';

export default {
  component: Table,
  tags: ['autodocs'],
  title: 'Table'
};

export const Default = {
  args: {
    cells: getCellsWithMockedData()
  }
};
