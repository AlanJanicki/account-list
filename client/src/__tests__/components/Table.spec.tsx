import React from 'react';

import { Table } from '../../components';
import { getCellsWithMockedData } from '../../mocks';
import { renderWithSnapshot } from '../common';

describe('<Table/>', () => {
  it('matches snapshot', () => renderWithSnapshot(<Table cells={getCellsWithMockedData()} />));
});
