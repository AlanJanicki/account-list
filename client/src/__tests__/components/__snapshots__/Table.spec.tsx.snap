// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  overflow: scroll;
}

.c0 span {
  padding: 1rem 0.3rem;
  border-bottom: 1px solid #dfdfdf;
}

.c0 span:nth-child(-n + 3) {
  border-bottom: 1px solid #000;
}

.c0 span:nth-last-child(-n + 3) {
  border-bottom: none;
}

@media (min-width:768px) {
  .c0 span {
    padding: 1.2rem 0.3rem;
  }
}

@media (min-width:992px) {
  .c0 span {
    font-size: 1.1rem;
  }
}

@media (min-width:768px) {
  .c0 span:nth-child(-n + 3) > * {
    font-size: 1.1rem;
  }
}

@media (min-width:992px) {
  .c0 span:nth-child(-n + 3) > * {
    font-size: 1.3rem;
  }
}

<div
    class="c0"
  >
    <span
      data-testid="header"
    >
      <strong>
        name
      </strong>
    </span>
    <span
      data-testid="header"
    >
      <strong>
        profitLoss
      </strong>
    </span>
    <span
      data-testid="header"
    >
      <strong>
        accountType
      </strong>
    </span>
    <span
      data-testid="cell"
    >
      Spread bet
    </span>
    <span
      data-testid="cell"
    >
      0.23
    </span>
    <span
      data-testid="cell"
    >
      IGSB
    </span>
    <span
      data-testid="cell"
    >
      New Spread bet
    </span>
    <span
      data-testid="cell"
    >
      -679
    </span>
    <span
      data-testid="cell"
    >
      IGSB
    </span>
    <span
      data-testid="cell"
    >
      Spread bet - demo
    </span>
    <span
      data-testid="cell"
    >
      16.211
    </span>
    <span
      data-testid="cell"
    >
      IGSB
    </span>
  </div>
</DocumentFragment>
`;
